### PR TITLE
HPCC-29136 Improve dfs getSize error if invalid compress part

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by shamser — CHANGES_REQUESTED** [review]

File: dali/base/dadfs.cpp [hunk 1]
```diff
@@ -7084,9 +7086,15 @@ offset_t CDistributedFilePart::getSize(bool checkCompressed)
...
-                if (compressedIO)
-                    ret = compressedIO->size();
+                Owned<IFileIO> partFileIO = partfile->open(IFOread);
+                if (partFileIO)
+                {
+                    Owned<ICompressedFileIO> compressedIO = createCompressedFileReader(partFileIO);
+                    if (compressedIO)
+                        ret = compressedIO->size();
+                    else
+                        throw new CDFS_Exception(DFSERR_PhysicalCompressedPartInvalid, firstname.str());;
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dadfs.cpp]: your right, fixed.

File: dali/base/dadfs.cpp [hunk 2]
```diff
@@ -314,6 +314,8 @@ class DECL_EXCEPTION CDFS_Exception: implements IDFS_Exception, public CInterfac
             return str.append(": Cluster does not have storage plane: ").append(errstr);
         case DFSERR_MissingStoragePlane:
             return str.append(": Storage plane missing: ").append(errstr);
+        case DFSERR_PhysicalCompressedPartInvalid:
+            return str.append(": compressed part is not in the valid format: ").append(errstr);
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/base/dadfs.cpp]: fixed

**Review by shamser — APPROVED** [review]

**Comment by ghalliday** [comment]
@jakesmith please squash.

**Comment by jakesmith** [comment]
@ghalliday - squashed

**Pull request merged**